src/test/scala/io/github/pauljamescleary/petstore/infrastructure/endpoint/UserEndpointsSpec.scala

```diff
@@ -69,4 +69,63 @@ class UserEndpointsSpec
         createdUser.id shouldEqual updatedUser.id
       }
     }
+
+  test("get user by userName") {
+    val userRepo = UserRepositoryInMemoryInterpreter[IO]()
+    val userValidation = UserValidationInterpreter[IO](userRepo)
+    val userService = UserService[IO](userRepo, userValidation)
+    val userHttpService: HttpService[IO] = UserEndpoints.endpoints[IO](userService)
+
+    implicit val userDecoder: EntityDecoder[IO, User] = jsonOf[IO, User]
+
+    val user = User("test", "test", "test", "test", "test", "test", None)
```

Comment:
The values are poor. If they were different it would catch the problem above with `firstName`